### PR TITLE
fix(kiwi): keep zoom_cap off the zoom_max start scale. Principle IV.

### LINKED ISSUE #5536: Kiwi waterfall start scale mismatch on servers with non-default zoom_max
The scale bug reported in #5528 is tracked separately here so #5529 can fix it independently of the Web-888 receiver-family work in #5530.

## Problem

The Kiwi client encodes waterfall start offsets with a fixed 2^24 scale. A compatible server advertising zoom_max=11 uses 1024 << 11 (2^21), so a request encoded with the larger scale can exceed its range and be clamped to the band edge. The reporter observed a misplaced or black/frozen waterfall when switching bands on a Web-888 receiver.

## Evidence and expected behavior

The reporter's capture in #5528 records the old request clamping to MAX_START(7)=2080768. This hardware observation is reporter-provided, not independently reproduced in the review.

The published RaspSDR server implementation derives HZperStart from WF_WIDTH << MAX_ZOOM:
https://github.com/RaspSDR/server/blob/68a64e1b39a3f291762904576f47c9d437fd3509/rx/rx_waterfall.cpp

Request encoding, quantization tolerance, and received frame start decoding should use the advertised zoom_max/zoom_cap. The default of 14 must retain the existing KiwiSDR 2^24 behavior. Late-arriving scale metadata must invalidate and resend the view using the corrected scale.

## Scope and validation

This issue covers scale conversion only. Family selection, WebSocket paths, binary MSG dispatch, and startup-burst handling remain in #5528/#5530.

PR #5529 supplies socket-free conversion tests. Independent review passed those and the existing protocol test on ARM64; restoring the old constant made the new scale test fail. Client resend ordering was inspected, but not exercised through a live server or a client transport-injection test.

### LINKED ISSUE #5655: Kiwi v1.900 shared waterfall: zoom_cap re-encodes the waterfall start on the wrong scale (even-noise waterfall)
## What happened

Selecting a KiwiSDR as a slice's RX antenna on current `main` shows a waterfall of even noise with no signals whenever the Kiwi gives the session a waterfall channel. Audio is fine. Channels the client treats as waterfall-less keep showing the radio's own waterfall, so they look "fine".

Seen on kphsdr.com:8072/:8073/:8074/:8075 and kiwisdr.kfsdr.com:8074–8076 (all KiwiSDR v1.900–1.902, `rx8.wf3`, `wf_share=1`). Regression since #5529 (commit 6701ffbc, closed #5536); v26.8.3 was fine on the same receivers.

## Cause

KiwiSDR v1.900+ shared waterfalls send `zoom_max=14` and then `zoom_cap=11` in the same W/F setup burst (`rx/rx_waterfall.cpp`: `"MSG wf_fft_size=1024 ... zoom_max=%d zoom_cap=%d ... wf_share=%d ..."`, `MAX_ZOOM, ZOOM_CAP`). The client stored both keys in one member (`m_waterfallZoomCap`, since #3668), and #5529 made that member the start fixed-point scale (`1024 << value`). The cap arrives last, so the `SET zoom=… start=…` request is re-encoded on a 2^21 scale while the server keeps reading it on 2^24 (`HZperStart = ui_srate / (WF_WIDTH << MAX_ZOOM)`). The frame header is decoded with the same 2^21 scale, so the wrong window is painted at the right place.

From `aethersdr-20260912-120327.log` (kphsdr.com:8075, slice at 14.270 MHz, 200 kHz view):

| Event | Request | Server window (30 MHz band) |
|---|---|---|
| after `zoom_max = 14` | `SET zoom=7 start=7914826` | 14.153–14.387 MHz |
| after `zoom_cap = 11` | `SET zoom=7 start=989353` | 1.769–2.003 MHz |

The server echoes `MSG zoom=7 start=989353` and serves the 1.8 MHz noise floor; the client labels it 14.153 MHz.

`aethersdr-20260818-183053.log` (same receiver, pre-#5529) shows the same `zoom_max`/`zoom_cap` pair with the start unchanged and echoed unclamped in the W/F header — above `MAX_START(7)` for a 2^21 scale — so the server reads `start` on 2^24 regardless of `zoom_cap`.

## Expected

`zoom_max` fixes the scale for request encoding and header decoding. `zoom_cap` only bounds the zoom a client may request and never changes the scale. Web-888 (`zoom_max=11`, no `zoom_cap`) keeps its 2^21 scale.

Fix incoming in a PR from this account.

### PR DESCRIPTION
**Pull request by quelleck**
## Summary

Fixes #5655. Regression from #5529 (commit 6701ffbc, which had resolved #5536); related #5528 / #5530. Follow-up filed as #5656.

Selecting a KiwiSDR as a slice's receive antenna shows a waterfall of even noise with no signals whenever the session gets a waterfall channel. Audio is unaffected. Channels the client treats as waterfall-less (`rx_chan >= wf_chans`) keep the radio's own waterfall and so look fine.

**Cause.** KiwiSDR v1.900+ shared waterfalls (`rx8.wf3`, `wf_share=1`, e.g. kphsdr.com and kiwisdr.kfsdr.com) advertise `zoom_max=14` and then `zoom_cap=11` in the same W/F setup burst (`rx/rx_waterfall.cpp`: `"MSG ... zoom_max=%d zoom_cap=%d ... wf_share=%d ..."`, `MAX_ZOOM, ZOOM_CAP`). Since #5529 the start fixed-point scale is `WF_WIDTH << zoom_max`, but the client stored both keys in one member (the single-key handler dates from #3668). The cap arrives last, so the `SET zoom=… start=…` request was re-encoded on a 2^21 scale while the server keeps reading it on 2^24 (`HZperStart = ui_srate_Hz / (WF_WIDTH << MAX_ZOOM)`). The frame header was decoded on the same 2^21 scale, so the wrong window was painted at the right place.

From `aethersdr-20260912-120327.log` against kphsdr.com:8075, slice at 14.270 MHz, 200 kHz view:

| Event | Request | Server window (30 MHz band) |
|---|---|---|
| after `zoom_max = 14` | `SET zoom=7 start=7914826` | 14.153–14.387 MHz |
| after `zoom_cap = 11` | `SET zoom=7 start=989353` | 1.769–2.003 MHz |

The server echoed `MSG zoom=7 start=989353` and served the 1.8 MHz noise floor; the client labeled it 14.153 MHz. `aethersdr-20260818-183053.log` (same receiver, build 26.8.3, before #5529) shows the same `zoom_max`/`zoom_cap` tokens with the start unchanged and echoed unclamped in the W/F header, above `MAX_START(7)` for a 2^21 scale, so the server reads `start` on 2^24 regardless of `zoom_cap`.

**Fix.** Keep the two values apart. `zoom_max` fixes the scale for request encoding and frame-header decoding. `zoom_cap` only bounds the zoom we request; one `effectiveWaterfallZoomCap()` (`min(cap, max)`) serves both the request loop and header-zoom validation, and the cap defaults to `zoom_max` when the server never sends one, so Web-888 (`zoom_max=11` alone) keeps its 2^21 scale. `zoom_cap` no longer invalidates the cached request, so an unchanged view is not re-sent when the cap arrives; the cap-from-server flag resets with the sockets so a reconnect starts from the new burst. Non-finite zoom values are ignored; finite values are bounded to [0, 20] before rounding or narrowing to avoid overflow.

Principle IV (clean-room): the scale rule comes from the public KiwiSDR server source (`rx/rx_waterfall.cpp`, `HZperStart` and the `wf_setup` message) plus the two wire captures above; the RaspSDR copy pinned in `KiwiSdrProtocol.h` matches.

## Changes

- `src/core/KiwiSdrClient.{h,cpp}`: separate `m_waterfallZoomMax` (scale) from `m_waterfallZoomCap` (request ceiling) with a `m_waterfallZoomCapFromServer` latch; handle the two MSG keys separately; constant renamed `kDefaultWaterfallZoomMax`; adds `friend class KiwiSdrWaterfallZoomCapTest` beside the existing setup-test friend.
- Beyond the minimal fix, added in response to review: a new `effectiveWaterfallZoomCap()` = `clamp(min(zoom_cap, zoom_max), 0, 20)` is now the single ceiling for both the `SET zoom` request loop and the frame-header zoom gate/decoder (`parseWaterfallFrameHeader`, `decodeWaterfallFrame`, and the two classify/decode calls in `handleWaterfallFrame`); previously the gate used whichever value the server sent last.
- Also from review: `zoom_max`/`zoom_cap` values are rejected when non-finite and bounded to [0, 20] before being rounded with `std::lround` instead of truncated (matching the existing `wf_cal` handling), `zoom_cap` no longer invalidates the cached request, and `zoom_max` invalidates it only when the value actually changed, so an unchanged setup burst no longer re-sends an identical `SET zoom` line. A `zoom_cap` that arrives before any server `zoom_max` records the cap but does not send; the `zoom_max` token that follows sends with the cap applied, so a server whose `zoom_max` is not 14 never sees one request on the seed scale (review nit).
- `tests/kiwi_sdr_waterfall_zoom_cap_test.cpp` (+ `tests/tests.cmake`): socket-free replay of the live v1.900 burst. Asserts the start stays at the server-echoed 7914826 and nothing is re-sent for the burst (scale and selected zoom unchanged); a 1040-byte frame whose 16-byte header echoes that start decodes to 14.153 MHz; `zoom_cap` caps a 1 kHz view at zoom 11 and the wire start, decoded on the server's scale, covers the view; a table of bursts (`zoom_max` first, `zoom_cap` first, `cap > max`, `zoom_cap` first on a `zoom_max=11` server, `zoom_max` only) checks scale, ceiling, re-send count and wire decoding for each; the header gate accepts a zoom at the effective ceiling (decoded on `zoom_max`) and drops one above it; a reconnect without `zoom_cap` releases the old ceiling (1 kHz view back to zoom 14, one re-send); `nan`/`inf` are ignored; `zoom_max=11` alone still drives the 2^21 scale with the wire start decoding into the 14.104 MHz view.
- `docs/kiwisdr-cleanroom-design.md` (the Kiwi Principle IV record named in AGENTS.md): the W/F `start` scale is now stated per server (`WF_WIDTH << zoom_max`, stale as a fixed 2^24 since #5529), `zoom_cap` is described as a request ceiling only, the 2026-09-12 observation is recorded, and §Source Provenance gains an entry for the v1.900 server-source facts (jks-prv/KiwiSDR at `3b570602`, LGPL-headed `rx/rx_waterfall.{h,cpp}` and `rx/rx_waterfall_cmd.cpp`).

The regression additionally covers both zoom keys at range and rounding boundaries and with finite values beyond integer and long ranges, checking that no invalid floating-point conversion occurs.

No CHANGELOG edits (maintainer writes it at release cut).

## Verification

- Final review build at `a8704220` succeeded natively on ARM64 with RADE enabled. An isolated native Cocoa demo smoke check connected `DEMO-0001`, tuned slice A to 14.270 MHz, confirmed Metal rendering and TX off, then disconnected. The review instances were stopped. This does not replace the author-provided live Kiwi proof below.

- Review repair on current main: the four focused socket-free targets (`kiwi_sdr_protocol_test`, `kiwi_sdr_waterfall_scale_test`, `kiwi_sdr_waterfall_setup_test`, and `kiwi_sdr_waterfall_zoom_cap_test`) pass on ARM64. Restoring the wrong request scale fails with `989353` versus `7914826`; independently restoring either post-conversion numeric clamp fails at `2147483648`; removing the test's preflight-interception observation fails its reset assertion. All mutations were restored, and all four tests passed again.
- The HTTP preflight is a protected virtual transport boundary with unchanged production behavior. The regression intercepts it before network manager/reply creation, asserts that the interception was reached on every connection reset, and checks both network pointers remain null. No synthetic radio peer or HTTP listener is used.
- Engine-boundary, test-registration, frozen CI-gate and touchpoint-manifest checks passed; no baseline or gate expansion.

- Author validation before the review repair: `ctest -R kiwi`, 13/13 passed. The final regression overrides the HTTP status-preflight entry point as well as the waterfall transport, and asserts that no network manager or reply was created. It does not rely on an unserviced main event loop to suppress Qt HTTP worker activity.
- Negative control on the first revision: with the `zoom_cap` handler temporarily re-conflated (cap also overwriting the scale member) the test failed at its first post-burst assertion (`zoom_max/zoom_cap were not recorded separately`); restored afterwards. The final test additionally decodes every wire start on the server's scale and checks the band it names covers the view, so a wrong-scale encoding fails on its own.
- Two adversarial review rounds (independent reviewers plus refutation passes) on the diff; every confirmed item was applied, no regressions found in the request dedup, header gate, or token-loop changes.
- **Author-provided live proof through the automation bridge before the review repair** (the original branch build, isolated `AETHER_SETTINGS_DIR`, demo simulator radio, slice tuned to 15.000 MHz, `slice rxsource "0 KPH8075"` → kphsdr.com:8075, KiwiSDR_1.900, receive channel 1, `wf_share=1`, uncompressed direct-bin rows, 0 sequence gaps):

  | View | Request sent | Server echo | Header `x_bin_server` | Row low edge (2^24) |
  |---|---|---|---|---|
  | 8 kHz (sim default) | `SET zoom=11 start=8382387` | `zoom=11 start=8382387` | `b3 e7 7f 00` = 8382387, zoom `0b` | 14.9889 MHz |
  | 200 kHz (zoomed out) | `SET zoom=7 start=8320947` | `zoom=7 start=8320947` | echoed | 14.8790 MHz |

  After `MSG zoom_cap = 11` no further `SET zoom` line was sent (the cap did not change the selected zoom). Every start is above 2^21, which a 2^21 server would have clamped to `MAX_START`, so the server is on 2^24. `get tracedebug 0`: `kiwiWaterfallActive=true`, 1624 finite Kiwi bins, peak −54 dBm on a −117 dBm floor. `get renderstats`: `waterfallSource=kiwi`, ~4.8 Kiwi updates/s, ~7 visible rows/s. Screenshot of the pan shows the WWV carrier exactly on 15.000 MHz in the Kiwi waterfall; on `main` the same session paints the 1.8 MHz noise floor there.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/kiwisdr-cleanroom-design.md [hunk 1]
```diff
@@ -846,8 +849,20 @@ headers echoing the requested 32-bit `start` plus one-byte `zoom`. A later
 receive-only browser WebSocket observation showed that the rendered page sends
 million-scale `start` values matching a 24-bit fixed-point low-edge offset
 within the full W/F bandwidth. AetherSDR therefore treats `start` as
-`round(((row_low - full_low) / full_bandwidth) * 2^24)`, not as a coarse
-segment index. The chosen zoom is the narrowest row span that covers the
+`round(((row_low - full_low) / full_bandwidth) * (WF_WIDTH << zoom_max))`,
+not as a coarse segment index. The scale is per server: the public KiwiSDR
+source (`rx/rx_waterfall.cpp`, `HZperStart = ui_srate_Hz / (WF_WIDTH <<
```
Comment (K5PTB) [inline docs/kiwisdr-cleanroom-design.md:871]: Nit: clean under Principle IV, but §"Source Provenance" records each server-source consult with repo, commit and licence header. The only `rx/rx_waterfall.cpp` entry is at `a83085fe`, for header layout and zoom limits. `ZOOM_CAP = kiwi.wf_share ? 11 : 14` and the v1.900 `wf_setup` line look newer than that snapshot; one bullet in the existing format would record which commit they came from.

File: src/core/KiwiSdrClient.cpp [hunk 2]
```diff
@@ -1876,7 +1893,7 @@ bool KiwiSdrClient::parseWaterfallFrameHeader(const QByteArray& frame,
 
     const quint32 parsedStart = readLittleEndianU32(frame.constData() + 4);
     const int parsedZoom = static_cast<uchar>(frame[8]);
-    if (parsedZoom < 0 || parsedZoom > m_waterfallZoomCap) {
+    if (parsedZoom < 0 || parsedZoom > effectiveWaterfallZoomCap()) {
```
Comment (K5PTB) [inline src/core/KiwiSdrClient.cpp:1898]: Nit: this gate is not independently pinned. Swapping `effectiveWaterfallZoomCap()` for `m_waterfallZoomMax` here passes all four Kiwi tests, because `decodeWaterfallFrame(frame, effectiveWaterfallZoomCap())` in `handleWaterfallFrame` drops the same above-ceiling frame. Redundant rather than wrong; the "header above the effective ceiling" leg just cannot tell the two checks apart.

**Review by K5PTB — COMMENTED** [review]
## Issue fit

**Yes.** #5655 names one defect: `zoom_cap` and `zoom_max` were stored in one member, so the cap re-encoded `start` on 2^21 while the server read 2^24. The PR splits them exactly as the issue's "Expected" section asks. Both scale sites now read `m_waterfallZoomMax`. Every ceiling site goes through `effectiveWaterfallZoomCap()`. The cap follows `zoom_max` unless the server sent one, which is what keeps Web-888 (`zoom_max=11` alone) on 2^21. I found no blockers. The fix and its test hold up under mutation, and the nits below are small.

## Scope

| File | What it changes | Claimed? | Verdict |
|---|---|---|---|
| `src/core/KiwiSdrClient.{h,cpp}` | Member split, `effectiveWaterfallZoomCap()`, separate `zoom_max`/`zoom_cap` handlers, non-finite guard, flag reset on teardown | Yes | In scope |
| `docs/kiwisdr-cleanroom-design.md` | Per-server scale rule, 2026-09-12 capture recorded | Yes | In scope; see the provenance nit |
| `tests/kiwi_sdr_waterfall_zoom_cap_test.cpp`, `tests/tests.cmake` | New registered regression test | Yes | In scope |

Everything in the diff is explained by the issue. No settings, capability or protocol surface is added. No `CHANGELOG.md` entry. The test binds, listens and connects nothing; the aethersdr-agent review already covers its unserviced status preflight. One bookkeeping note: the body's "closed #5536" is parsed as a closing keyword, so the PR lists #5536, already closed, among the issues it closes. Harmless.

## Blockers

None.

## Nits (non-blocking)

- **Provenance entry for the new server-source facts** (inline on `docs/kiwisdr-cleanroom-design.md:854`). Principle IV lists open-source references as clean, so the source itself is fine. But this doc records every KiwiSDR server consult in §"Source Provenance", with repo, commit and licence header. The only recorded consult of `rx/rx_waterfall.cpp` is at `a83085fe`, for header layout and zoom limits. `ZOOM_CAP = kiwi.wf_share ? 11 : 14` and the `zoom_max=%d zoom_cap=%d` setup line are v1.900 facts that a 2026-06-28 snapshot may predate. One bullet in the existing format would keep the record complete.
- **The header-zoom gate in `parseWaterfallFrameHeader` is not independently pinned** (inline on `KiwiSdrClient.cpp:1896`). Changing it from `effectiveWaterfallZoomCap()` to `m_waterfallZoomMax` passes the whole Kiwi suite. The "header above the effective ceiling was accepted" leg still goes green because `decodeWaterfallFrame(frame, effectiveWaterfallZoomCap())` drops the same frame one step later. That makes the gate redundant rather than wrong, but the test cannot tell the two checks apart.
- I agree with both aethersdr-agent nits: the transient stale-scale send when `zoom_cap` precedes `zoom_max`, and the "Socket-free" wording. I have nothing to add to them.

## What I tried to break

- **Ran 9 mutations** of `KiwiSdrClient.cpp` against `kiwi_sdr_waterfall_zoom_cap_test`, `kiwi_sdr_waterfall_setup_test`, `kiwi_sdr_waterfall_scale_test` and `kiwi_sdr_protocol_test`. The new test caught 8:
  - request encoding scaled on the cap
  - header decoding scaled on the cap
  - `zoom_cap` invalidating the cached request
  - an unchanged `zoom_max` invalidating it
  - no `isfinite` guard on `zoom_cap`
  - the ceiling ignoring `zoom_max`
  - `cleanupSockets` keeping the cap flag
  - the cap always following `zoom_max`

  The one survivor is the gate nit above. The existing Kiwi tests caught none of the 8, so this test is the only coverage for the fix.
- **Checked the new `continue`s.** They sit inside the per-token `else if` chain in `handleTextMessage`, so a non-finite value skips only its own token. That matches the existing unparsable-value path.
- **Recomputed the logged starts.** `round((14.1528125 / 30) × 2^24) = 7914826` and `round((14.1528125 / 30) × 2^21) = 989353`, matching both rows of the #5655 table.
- **Built and ran all four Kiwi tests at 29ec765b.** All pass.

**Not driven:** the automation bridge. The body's live proof runs against kphsdr.com, a shared public receiver with limited channels, and the demo simulator has no Kiwi server to exercise this path. I have not re-run it against a third party's receiver.

File: src/core/KiwiSdrClient.cpp [hunk 3]
```diff
@@ -2596,10 +2614,37 @@ void KiwiSdrClient::handleTextMessage(StreamKind stream, const QString& text)
...
+            // scale: the server keeps reading start on WF_WIDTH << zoom_max.
+            // The cached request stays valid: the view is recomputed with
+            // the new ceiling and only re-sent when the zoom/start changed.
+            if (!std::isfinite(value)) {
+                continue;
+            }
+            m_waterfallZoomCap = std::clamp(
+                static_cast<int>(std::lround(value)), 0, 20);
+            m_waterfallZoomCapFromServer = true;
             sendWaterfallViewToServer();
```
[the remark this answers is not in the document]
Comment (quelleck) [reply src/core/KiwiSdrClient.cpp]: Applied in 7c9343b2. A `zoom_cap` that arrives before any server `zoom_max` now records the cap and does not send; the `zoom_max` token that follows sends with the cap applied (`m_waterfallZoomMaxFromServer`, reset with the other flag). The burst table gained a `zoom_cap=11 zoom_max=11` row that pins exactly one request, decoded on 2^21; with the deferral removed that row sends two (the first on 2^24) and the test fails, so it is not vacuous.

File: tests/kiwi_sdr_waterfall_zoom_cap_test.cpp [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+// wf_share=1) advertises "zoom_max=14" and then "zoom_cap=11" in the same
+// W/F setup burst (rx/rx_waterfall.cpp: "MSG ... zoom_max=%d zoom_cap=%d ...
+// wf_share=%d", MAX_ZOOM, ZOOM_CAP). The server's start fixed-point scale is
+// WF_WIDTH << MAX_ZOOM (HZperStart = ui_srate_Hz / (WF_WIDTH << MAX_ZOOM));
+// zoom_cap only bounds the zoom a client may request. Treating zoom_cap as
+// the scale re-encoded a 14.153 MHz request as start=989353 on a 2^21 scale,
+// which the server read on 2^24 as 1.769 MHz, and the frame header decoded
+// with the same 2^21 scale labeled that noise as 14.153 MHz: a waterfall of
+// even noise with no signals (kphsdr.com:8075, 2026-09-12, start 7914826 ->
+// 989353 after "zoom_cap = 11"). Socket-free: the transport seam is injected,
```
[the remark this answers is not in the document]
Comment (quelleck) [reply tests/kiwi_sdr_waterfall_zoom_cap_test.cpp]: Applied in 7c9343b2. The header now reads: no socket is bound, listened to or connected; the transport seam is injected, frames enter through `handleWaterfallFrame()`, and the status preflight that `connectToEndpoint()` schedules is never serviced (no event loop runs).

File: docs/kiwisdr-cleanroom-design.md (hunk 1 above)
Comment (quelleck) [reply docs/kiwisdr-cleanroom-design.md:871]: Applied in 7c9343b2: a §Source Provenance bullet in the existing format for `https://github.com/jks-prv/KiwiSDR.git` at `3b570602a14161188f8c85eede62ab2a70e7ad66` (the consulted `rx/rx_waterfall.h`, `rx/rx_waterfall.cpp` and `rx/rx_waterfall_cmd.cpp` were re-fetched at that commit and are byte-identical to the copies used). It records `ZOOM_CAP = kiwi.wf_share ? 11 : 14` (split from `MAX_ZOOM` in upstream b71558744af5, 2026-04-30), the `wf_setup` line with `zoom_max=MAX_ZOOM zoom_cap=ZOOM_CAP ... wf_share=`, `HZperStart = ui_srate_Hz / (WF_WIDTH << MAX_ZOOM)`, and the `SET zoom=` clamp to `ZOOM_CAP`; all three files carry the LGPL 2-or-later header in that snapshot. It also notes that `rx/rx_util.cpp` and `web/openwebrx/openwebrx.js` were displayed for the `wf_share` availability gate (#5656) and were not used as references for this change.

File: src/core/KiwiSdrClient.cpp (hunk 2 above)
Comment (quelleck) [reply src/core/KiwiSdrClient.cpp:1898]: Agreed that it is redundant rather than wrong, and I left it as is. The two checks cannot be told apart by any test because they have no observable difference: an extended header whose zoom exceeds the ceiling always leaves a 1040 − 4 = 1036-byte payload once `inspectWaterfallFrame` refuses the header, so `decodeWaterfallFrame` rejects the frame whether or not `parseWaterfallFrameHeader` had already declined it, and the row is dropped either way. The parse gate stays so that `parseWaterfallFrameHeader` never reports a zoom the client will refuse (it is a self-contained parser), and both sites read the same `effectiveWaterfallZoomCap()` so they cannot drift. Folding the two into one would mean restructuring `handleWaterfallFrame` around the decode result, which is beyond this fix.

**Review by rfoust — APPROVED** [review]
The fix addresses #5655: request encoding and header decoding use zoom_max, while zoom_cap only limits the requested zoom. The reported test-isolation blocker and numeric-conversion nit are fixed in 3c55cecc; current main ead241fb is incorporated in reviewed head a8704220.

### Scope
| Files | Change | Verdict |
|---|---|---|
| KiwiSdrClient.h/.cpp | Separate scale/ceiling, inject HTTP preflight for tests, bound zoom before numeric conversion | In scope |
| Waterfall zoom-cap test and tests.cmake | Registered socket-free regression, transport interception assertions, large finite and rounding boundary cases | In scope |
| Kiwi clean-room design note | Scale contract and pinned source provenance | In scope |

No remaining blockers or nits. No unrelated behavior, new settings, or CI-gate expansion.

### Verification
- ARM64 application build succeeded on a8704220 with RADE enabled; host/system processors and executable are ARM64 and the build graph contains no RNNoise x86 sources.
- All four focused tests passed on the final tree: kiwi_sdr_protocol_test, kiwi_sdr_waterfall_scale_test, kiwi_sdr_waterfall_setup_test, kiwi_sdr_waterfall_zoom_cap_test.
- Four negative controls failed as expected: request scale restored to zoom_cap (989353 instead of 7914826); overflow-prone conversion restored separately for zoom_max and zoom_cap (2147483648 becomes 0); missing preflight-interception observation. Restored the sources and reran all four successfully. The HTTP transport stayed intercepted during every mutation.
- Static engine-boundary, test-registration, frozen CI-gate, and touchpoint-manifest checks passed.
- Native Cocoa automation smoke check on this build, isolated settings and AETHER_AUTOMATION_NO_TX=1: DEMO-0001 connected, slice A read frequency=14.27, rendering used Metal, and disconnect read connected=false/transmitting=false. Both review processes were stopped. Offscreen startup did not expose the bridge; Cocoa did. This is an application smoke check, not new live Kiwi-receiver evidence; the author-provided live proof remains separately attributed in the PR body.
- Independently verified the scale contract in the pinned KiwiSDR server source. Inspected cap ordering, Web-888 fallback, reconnect reset, header gating, and deduplication.

Approved as requested by the operator. Platform CI is still running and must pass before squash merge.
